packages/fluentui/react-northstar/src/components/Table/TableRow.tsx

```diff
@@ -52,94 +52,111 @@ const handleVariablesOverrides = variables => predefinedProps => ({
-      content: false,
-    }),
-    content: customPropTypes.every([
-      customPropTypes.disallow(['children']),
-      PropTypes.oneOfType([PropTypes.arrayOf(customPropTypes.nodeContent), customPropTypes.nodeContent]),
-    ]),
-    items: customPropTypes.collectionShorthand,
-    header: PropTypes.bool,
-    compact: PropTypes.bool,
-    selected: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    accessibility: tableRowBehavior as Accessibility,
-  };
-
-  rowRef = React.createRef<HTMLElement>();
-
-  actionHandlers = {
-    // https://github.com/microsoft/fluent-ui-react/issues/2150
-    unsetRowTabbable: e => {
-      this.rowRef.current.setAttribute('tabindex', '-1');
-    },
-    performClick: e => {
-      this.handleClick(e);
+export type TableRowStylesProps = Pick<TableRowProps, 'header' | 'compact'>;
+
+const TableRow: React.FC<WithAsProp<TableRowProps>> & FluentComponentStaticProps<TableRowProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(TableRow.displayName, context.telemetry);
+  setStart();
+  const rowRef = React.useRef<HTMLElement>();
+  const { className, design, styles, items, header, compact, children, accessibility, variables, selected } = props;
+
+  const hasChildren = childrenExist(children);
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(TableRow.handledProps, props);
+  const getA11yProps = useAccessibility(accessibility, {
+    debugName: TableRow.displayName,
+    actionHandlers: {
+      // https://github.com/microsoft/fluent-ui-react/issues/2150
+      unsetRowTabbable: e => {
+        rowRef.current.setAttribute('tabindex', '-1');
+      },
+      performClick: e => {
+        handleClick(e);
+      },
     },
-  };
+    mapPropsToBehavior: () => ({
+      selected,
+      header,
+    }),
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<TableRowStylesProps>(TableRow.displayName, {
+    className: tableRowClassName,
+    mapPropsToStyles: () => ({
+      header,
+      compact,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
+  const handleClick = (e: React.SyntheticEvent) => {
     if (e.currentTarget === e.target) {
-      _.invoke(this.props, 'onClick', e, this.props);
+      _.invoke(props, 'onClick', e, props);
       e.preventDefault();
     }
   };
 
-  renderCells(accessibility: ReactAccessibilityBehavior, variables: ComponentVariablesObject) {
-    const { items } = this.props;
-
-    const cellAccessibility = accessibility.childBehaviors ? accessibility.childBehaviors.cell : undefined;
+  const renderCells = () => {
+    // TODO: consider moving to useAccessibility()
+    const childBehavior = accessibility && accessibility({ header, selected }).childBehaviors;
```

Comment:
I'm not sure how else to access childBehavior definition from the `accessibility` prop

Reply:
Agreed to create a separate PR with useAccessibility update